src/librustdoc/doctest.rs

```diff
@@ -686,75 +796,142 @@ impl CreateRunnableDoctests {
         let test_id = format!(
             "{file}_{line}_{number}",
             file = file,
-            line = test.line,
+            line = scraped_test.line,
             number = {
                 // Increases the current test number, if this file already
                 // exists or it creates a new entry with a test number of 0.
                 self.visited_tests
-                    .entry((file.clone(), test.line))
+                    .entry((file.clone(), scraped_test.line))
                     .and_modify(|v| *v += 1)
                     .or_insert(0)
             },
         );
 
-        let rustdoc_options = self.rustdoc_options.clone();
-        let rustdoc_test_options = IndividualTestOptions::new(&self.rustdoc_options, test_id, path);
-
-        debug!("creating test {name}: {}", test.text);
-        self.tests.push(test::TestDescAndFn {
-            desc: test::TestDesc {
-                name: test::DynTestName(name),
-                ignore: match test.langstr.ignore {
-                    Ignore::All => true,
-                    Ignore::None => false,
-                    Ignore::Some(ref ignores) => ignores.iter().any(|s| target_str.contains(s)),
-                },
-                ignore_message: None,
-                source_file: "",
-                start_line: 0,
-                start_col: 0,
-                end_line: 0,
-                end_col: 0,
-                // compiler failures are test failures
-                should_panic: test::ShouldPanic::No,
-                compile_fail: test.langstr.compile_fail,
-                no_run: test.no_run(&rustdoc_options),
-                test_type: test::TestType::DocTest,
+        let edition = scraped_test.edition(&self.rustdoc_options);
+        let doctest = DocTest::new(
+            &scraped_test.text,
+            Some(&self.opts.crate_name),
+            edition,
+            self.can_merge_doctests,
+            Some(test_id),
+        );
+        let is_standalone = !self.can_merge_doctests
+            || !doctest.can_be_merged
```

Comment:
What's the difference between `!doctest.can_be_merged` and `is_standalone`? Why do we need both?

Reply:
A lot of doctests cannot be merged for obvious reasons (`compile_fail` is a good example). So `is_standalone` is "this doctest should only be run on its own".

Reply:
`self.can_merge_doctests` comes from `edition >= 2024`, so not exactly the same. Might be worth a comment.

Reply:
So to clarify:

- `self.can_merge_doctests` -> means overall crate edition >= 2024
- (notice !) `!doctest.can_be_merged` -> means this particular doctest failed to be parsed or has some other syntactic issue (e.g. `$crate`)
- all the others are various features that disqualify merging

Is this right? If so, I still think we should remove all these ORs here and compute them as part of `can_be_merged`. It seems confusing to separate syntactic issues from other things that disqualify merging.